website/docs/configuration/providers.html.md

```diff
@@ -38,79 +35,44 @@ provider "google" {
 ```
 
 The name given in the block header (`"google"` in this example) is the
-[local name](./provider-requirements.html#local-names) of the provider
-to configure.
-
-The body of the block (between `{` and `}`) contains configuration arguments
-for the provider itself. Most arguments in this section are defined by
-the provider itself; in this example both `project` and `region`
-are specific to the `google` provider.
-
-The configuration arguments defined by the provider may be assigned using
-[expressions](./expressions.html), which can for example
-allow them to be parameterized by input variables. However, since provider
-configurations must be evaluated in order to perform any resource type action,
-provider configurations may refer only to values that are known before
-the configuration is applied. In particular, avoid referring to attributes
-exported by other resources unless their values are specified directly in the
-configuration.
+[local name](./provider-requirements.html#local-names) of the provider to
+configure. This provider should already be included in a `required_providers`
+block.
+
+The body of the block (between `{` and `}`) contains configuration arguments for
+the provider. Most arguments in this section are defined by the provider itself;
+in this example both `project` and `region` are specific to the `google`
+provider.
+
+You can use [expressions](./expressions.html) in the values of these
+configuration arguments, but can only reference values that are known before the
+configuration is applied. This means you can safely reference input variables,
+but not attributes exported by resources (with an exception for resource
+arguments that are specified directly in the configuration).
+
+A provider's documentation should list which configuration arguments it expects.
+For providers distributed on the
+[Terraform Registry](https://registry.terraform.io), versioned documentation is
+available on each provider's page, via the "Documentation" link in the
+provider's header.
+
+Some providers can use shell environment variables (or other alternate sources,
+like VM instance profiles) as values for some of their arguments; when
+available, we recommend using this as a way to keep credentials out of your
+version-controlled Terraform code.
 
 There are also two "meta-arguments" that are defined by Terraform itself
 and available for all `provider` blocks:
 
-- [`version`, for constraining the allowed provider versions][inpage-versions]
 - [`alias`, for using the same provider with different configurations for different resources][inpage-alias]
+- [`version`, which we no longer recommend][inpage-versions] (use
```

Comment:
Yeah... if I only linked the name of the argument, it would look like one of those fake-out go-nowhere links. 😑 Which is exactly what I did in the modules page, sigh. Fix one issue to make the other one worse.